discovery_engine/lib/src/domain/engine/engine.dart

```diff
@@ -23,9 +25,14 @@ import 'package:xayn_discovery_engine/src/domain/models/unique_id.dart'
 
 /// Interface to Discovery Engine core.
 abstract class Engine {
+  /// Configuration needed to initialize the engine.
+  final EngineConfig config;
```

Comment:
Why we want to store it here? It will be a parameter of the constructor of the real implementation but once we pass it to rust we can forget about it.